src/frontend/src/lib/types/qr-code.ts

```diff
@@ -6,3 +6,28 @@ export type QrResponse = {
 	token?: string;
 	amount?: number;
 };
+
+type DecodedUrnBase = {
```

Comment:
should we use zod to parse the decoded values to ensure those have the expected format?